### PR TITLE
feat(search): add support for delete events in search service #141

### PR DESCRIPTION
**Pull request by HoucemKacem**

### CODE REVIEW

File: search-service/src/main/kotlin/com/egm/stellio/search/service/EntityEventListenerService.kt [hunk 1]
```diff
@@ -76,6 +69,59 @@ class EntityEventListenerService(
             logger.error(RECEIVED_NON_PARSEABLE_ENTITY, e)
         }
 
+    private fun handleEntityDeleteEvent(entityDeleteEvent: EntityDeleteEvent) =
+        temporalEntityService.deleteTemporalEntityReferences(
```
Comment (bobeal) [inline search-service/src/main/kotlin/com/egm/stellio/search/service/EntityEventListenerService.kt:73]: it misses some consistency: you don't use the same service when creating the temporal entity references (`temporalEntityAttributeService`) and when deleting them (`temporalEntityService`)

File: search-service/src/main/kotlin/com/egm/stellio/search/service/AttributeInstanceService.kt [hunk 2]
```diff
@@ -127,6 +128,53 @@ class AttributeInstanceService(
...
+            .bind("entity_id", entityId)
+            .bind("attribute_name", attributeName)
+            .let {
+                if (datasetId != null) it.bind("dataset_id", datasetId)
+                else it
+            }
+            .fetch()
+            .rowsUpdated()
+
+    fun deleteAttributeInstancesOfAllInstancesOfTemporalAttribute(entityId: URI, attributeName: String): Mono<Int> =
```
Comment (bobeal) [inline search-service/src/main/kotlin/com/egm/stellio/search/service/AttributeInstanceService.kt:163]: `deleteAllAttributeInstancesOfTemporalAttribute`?

Comment (HoucemKacem) [reply search-service/src/main/kotlin/com/egm/stellio/search/service/AttributeInstanceService.kt]: Yes, updated

File: search-service/src/main/kotlin/com/egm/stellio/search/service/EntityEventListenerService.kt (hunk 1 above)
Comment (HoucemKacem) [reply search-service/src/main/kotlin/com/egm/stellio/search/service/EntityEventListenerService.kt]: Updated [here](https://github.com/stellio-hub/stellio-context-broker/pull/398/commits/b73f9522e9199a2ed33196a54ce916a74bee112f).

**Review by bobeal — APPROVED** [review]

**Pull request merged**